apps/health_campaign_field_worker_app/lib/main.dart

```diff
@@ -52,72 +71,84 @@ class MainApplication extends StatelessWidget {
+              actionType: 'search',
+            )),
         ),
         BlocProvider(
           create: (context) =>
               TableHideActionBloc(const TableHideActionState()),
         ),
       ],
-      child: BlocBuilder<ApplicationConfigBloc, ApplicationConfigState>(
+      child: BlocBuilder<AppInitializationBloc, AppInitializationState>(
         builder: (context, appConfigState) {
           return BlocBuilder<AuthBloc, AuthState>(
             builder: (context, authState) {
-              final appConfig =
-                  appConfigState.appConfigDetail?.configuration?.appConfig;
-              final tenantId = appConfigState.appConfigDetail?.tenantId;
+              final appConfig = appConfigState.appConiguration;
+              // final tenantId = appConfigState.appConiguration?;
               final localizationModulesList = appConfig?.localizationModules;
+              final firstLanguage = appConfig?.languages?.first.value;
+              final languages = appConfig?.languages;
 
-              return BlocProvider(
-                create: (appConfig != null && localizationModulesList != null)
-                    ? (context) => LocalizationBloc(
-                          const LocalizationState(),
-                          LocalizationRepository(client),
-                        )..add(LocalizationEvent.onLoadLocalization(
-                            module: localizationModulesList
-                                .map((e) => e.value.toString())
-                                .join(',')
-                                .toString(),
-                            tenantId: tenantId ?? "default",
-                            locale: appConfig.languages.first.value,
-                          ))
-                    : (context) => LocalizationBloc(
-                          const LocalizationState(),
-                          LocalizationRepository(client),
-                        ),
-                child: MaterialApp.router(
-                  supportedLocales: appConfig != null
-                      ? appConfig.languages.map((e) {
-                          final results = e.value.split('_');
+              return appConfig != null
+                  ? BlocProvider(
+                      create: (localizationModulesList != null &&
+                              firstLanguage != null)
+                          ? (context) => LocalizationBloc(
+                                const LocalizationState(),
+                                LocalizationRepository(client, isar),
+                                isar,
+                              )..add(LocalizationEvent.onLoadLocalization(
+                                  module: localizationModulesList
+                                      .map((e) => e.value.toString())
+                                      .join(',')
+                                      .toString(),
+                                  tenantId: "pb",
```

Comment:
hard coded tenantId

Reply:
@ajilo-eGov  Localisation values where not added to default tenant hence for testing we have hardcoded pb

Reply:
we should still read the tenantId to be used by the front-end app from config and default to "default" if not found. We can add a new property to the field-app-config MDMS for this.